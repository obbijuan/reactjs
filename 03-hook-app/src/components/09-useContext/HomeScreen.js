import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const HomeScreen = () => {

    const { user } = useContext(UserContext)

    return (
        <div>
            <h1>HomeScreen</h1>
            <br />
            <pre>{ JSON.stringify(user, null, 2)}</pre>
        </div>
    )
}
