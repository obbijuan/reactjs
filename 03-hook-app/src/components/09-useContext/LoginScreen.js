import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const { setUser } = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <br />
            <button
                className="btn btn-primary"
                onClick={() => setUser({
                    id: 13,
                    user: 'john'
                })}
            >
                login
            </button>
        </div>
    )
}
