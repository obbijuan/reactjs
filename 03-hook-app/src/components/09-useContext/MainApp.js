import React from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './UserContext'

export const MainApp = () => {

    const user = {
        id: 123,
        name: 'John',
        email: 'john.g@gmail.com'
    }

    return (
        <UserContext.Provider value={ user }>
            <AppRouter/>
        </UserContext.Provider>
    )
}
