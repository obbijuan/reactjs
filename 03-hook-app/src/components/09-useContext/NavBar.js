import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-sm navbar-light bg-light">
            <Link to="/" className="navbar-brand"> useContext </Link>
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    {/* El NavLink puede establecer una clase si el url coincide con el path */}
                    <NavLink exact to="/" className="nav-item nav-link active"> Home </NavLink>
                    <NavLink exact to="/about" className="nav-item nav-link"> About </NavLink>
                    <NavLink exact to="/login" className="nav-item nav-link"> Login </NavLink>
                </div>
            </div>
        </nav>
    )
}
