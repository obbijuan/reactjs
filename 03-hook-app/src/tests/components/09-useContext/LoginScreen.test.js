import React from 'react';
import { mount } from 'enzyme';
import { LoginScreen } from '../../../components/09-useContext/LoginScreen';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('pruebas en <LoginScreen/>', () => {

    const setUser = jest.fn();
    // wrapper //mount
    const wrapper = mount(
        <UserContext.Provider value={{
            setUser
        }}>
            <LoginScreen/>
        </UserContext.Provider>
    )
    
    test('debe de mostrarse correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    })
    
    test('debe ejecutar el setUser con el argumento esperado', () => {

        wrapper.find('button').prop('onClick')();
        expect(setUser).toHaveBeenCalledWith({
            id: 13,
            user: 'john'
        });
        
    })
    

})
