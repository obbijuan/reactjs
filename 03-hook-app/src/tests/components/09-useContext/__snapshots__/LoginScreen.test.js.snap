// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <LoginScreen/> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    LoginScreen
  </h1>
  <br />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    login
  </button>
</div>
`;
